### DISCUSSION TITLE
GPU Memory Utilization Percentage

### DISCUSSION THREAD

**Discussion by MagFlux** [LibreHardwareMonitor/LibreHardwareMonitor#887] [Ideas]
Just an idea as it seems this was actually removed from the fork parent OHM but OHM has GPU Memory Utilization as a percentage under the "Load" category for GPUs, would be nice to have this info back with LHM as it's own discrete line item.

Anyhow, great job on all the work the team has done, glad to see you all have taken OHM and continued building upon it!

**Comment by ntoff** [comment]
GPU memory util is there under "Data" for me. They're grey because I chose to hide them, perhaps you did too at one point? Try checking view -> show hidden sensors

![image](https://user-images.githubusercontent.com/9030085/202877859-2d841dc1-3eac-4a20-8fb0-9d064cbadf1d.png)

**Reply by MagFlux** [reply]
Hey @ntoff, thanks for the reply and yes I too have that info but if you look at my original post, I'm specifically asking the percent used to be displayed as a line item (i.e. (GPU Memory Free / GPU Memory Used) * 100).

For example OHM has GPU Memory Free in MB, GPU Memory Used in MB, AND GPU Memory as a %:
![image](https://user-images.githubusercontent.com/26913665/202925662-7ef821df-4ee9-4954-8205-ac3906270ea8.png)

Versus what LHM has below (note GPU Memory Controller is not memory utlilization):
![image](https://user-images.githubusercontent.com/26913665/202925704-7b271c2e-bef9-40f8-bf6a-f3ec609707a5.png)

**Comment by ntoff** [comment]
oh my bad, misinterpreted what you were asking

**Comment by ntoff** [comment]
@PhyxionNL do you have any opinions on this? I added https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/commit/aab811f5a7edec8144e405b724ef16f02ed9c9f0 which gives the memory usage as a percentage

![image](https://user-images.githubusercontent.com/9030085/203666884-273374dd-14ce-4431-b4ff-a0d4f397f89d.png)

I took the name from the "Data" section "GPU Memory Used" to hopefully be clear enough as to what this value is vs "GPU Memory Controller" (which is the Framebuffer I think)

I don't have a way to test or add this for AMD or Intel GPUs though so is this worth a pull request? I do see that the AMD GPU code already contains a `_memoryLoad` variable so I assume AMD GPUs already have this feature?

**Reply by PhyxionNL** [reply]
That would be fine! Although the name could just be GPU Memory I think 🙂

**Reply by ntoff** [reply]
One thing I'm not entirely sure I'm doing properly is the calculation part. The values for free/total are defined as uint but needed to be floats in order for the calculation to work.

`_memoryLoad.Value = ((float)(total - free) / total) * 100;` was what I came up with, but 
` _memoryLoad.Value = _memoryUsed.Value / _memoryTotal.Value * 100;` also works, but I assume is an implicit cast to a float? 

Which method would you prefer? I went with the first one just so all the calculations were done on the original variables (`free` and `total`) but I don't know if doing it the 2nd way would be "better"?

**Comment by MagFlux** [comment]
@ntoff Hey! Just wanted to say thanks for actually following up this! Also apologies, I realize now the first time around I used the wrong equation for my example.
